RFD.md

```diff
@@ -0,0 +1,191 @@
+- `rps stop -h <host> <job_id>`: Stop a running job with the given job_id
+
+The host can also be specified via the environment variable `RPS_HOST` to avoid having to specify it as CLI argument.
+
+`job_id` is a unique identifier for a job, implemented as a UUID.
+
+Here are some examples of how the CLI will work:
+
+```sh
+$ rps connect 158.101.135.237
+Connected to Remote Linux Process Manager at 158.101.135.237
+(enter "help" for usage)
+> rps start -c "for i in {1..10}; do echo $i; sleep 1; done"
+Job started with job_id: 12345678-1234-1234-1234-123456789012
+> rps status 12345678-1234-1234-1234-123456789012
+Job 12345678-1234-1234-1234-123456789012 is running
+- started: 2025-04-20 12:31:45 -0500 CDT
+- command: "for i in {1..10}; do echo $i; sleep 1; done"
+> rps output 12345678-1234-1234-1234-123456789012
+1
+2
+3
+4
+> rps stop 12345678-1234-1234-1234-123456789012
+Job 12345678-1234-1234-1234-123456789012 is stopped
+```
+
+### API
+
+#### Authentication
+
+Authentication will be handled using mTLS. TLS 1.3 with `TLS_AES_256_GCM_SHA384` cipher suite will be used for maximum security.
+
+For simplicity, the certificates (CA, server, client) will be generated using `openssl` with the `Ed25519` algorithm and checked into source. In a real application, the certificates should be generated and stored in a secure manner.
+
+#### Authorization
+
+The server will use the client certificate's `Subject.CommonName` as the `clientId` to identify the client.
+
+Authorization will be segmented into:
+
+- `rpsreader`: read-only access (`GetJobStatus`, `StreamJobOutput`)
+- `rpsmanager`: full job management access (read + `StartJob` & `StopJob`)
+
+Users (clients) will be assigned one of these two roles in a hard-coded map for simplicity. In a real application, this assignment would be handled by an authorization server with groups and permissions.
+
+#### GRPC
+
+The API will be a gRPC server that handles messages from the CLI. Protocol Buffers (protobuf) will be used to define and serialize the methods and messages, which will provide consistency between the gRPC server and client.
+
+### Library
+
+The library will implement the business logic for the API, interacting with the linux system to start, stop, and manage jobs.
+
+#### Lifecycle
+
+0. Initialize (system setup)
+    - cgroups are initialized with the specified resource limits
+    - SIGCHLD handler is set up to handle process termination (asynchronously)
+1. Start Job
+    - process is assigned a job ID (UUID) and added to an in-memory process map
+    - output file is created, identified by the job ID
+    - process is started as a child process in user mode under `rpsuser`
+      - process output is redirected to the output file on process start
+      - process is added to the cgroup on process start using the linux `cgexec` utility, which will allow running a process directly within a specified control group without having to manually add it
```

Comment:
Try not to use a external tool like `cgexec`, can you implement it yourself in Go?

Reply:
switched to use the `CLONE_NEWCGROUP` flag